### DISCUSSION TITLE
How to iterate over classpath or how to unwrap T?

### DISCUSSION THREAD

**Discussion by Sailsman63** [com-lihaoyi/mill#1782] [Q&A] [answered]
Working on trying to construct a staging directory for a full application. I need to place all runtime dependencies in a specific folder, call it `T.dest / "lib"`.

`runClasspath` is of type `[T[Seq[PathRef]]]`. I can't figure out how to access that internal `Seq` for the purpose of iterating over it and copying each jar to `lib/`

Or is there a more idiomatic way of doing this?

**Comment by lefou** [comment, marked as answer]
```scala
def copyToLib(): Command[Unit] = T.command {
  runClasspath()
    .map(_.path) // the path
    .filter(p => os.exists(p) && os.isFile(p)) // only existing files
    .foreach(p => os.copy.over(p, millSourcePath / "lib" / p.last)) // copy files flat into lib/
}
```

**Reply by Sailsman63** [reply]
Thanks! It turns out that the part I was missing was the difference between `runClasspath` and `runClasspath()`. Probably comes down to me being a total Scala noob.
